odoo/tools/template_inheritance.py

```diff
@@ -205,23 +205,94 @@ def extract(spec):
                     raise ValueError(_("Invalid mode attribute:") + " '%s'" % mode)
             elif pos == 'attributes':
                 for child in spec.getiterator('attribute'):
-                    attribute = child.get('name')
-                    value = child.text or ''
-                    if child.get('add') or child.get('remove'):
-                        assert not child.text
-                        separator = child.get('separator', ',')
-                        if separator == ' ':
-                            separator = None    # squash spaces
-                        to_add = (
-                            s for s in (s.strip() for s in child.get('add', '').split(separator))
-                            if s
+                    attribute = child.attrib.pop('name')
+                    value = None
+
+                    if child.get('add') or child.get('remove') or 'separator' in child.attrib:
```

Comment:
This condition is way too sloppy.  What happens if you only have attribute `separator`?  Nothing, no error.  There should be a test with only `separator`, which IMHO should raise an exception.
I suggest to be more specific: the element `<attribute>` should only have attributes:
- `name` (mandatory),
- either `add`, `remove` (both with optional `separator`) or none of them,
- any attribute that starts with `data-oe-`